pex/pip/tool.py

```diff
@@ -393,6 +398,34 @@ def _spawn_pip_isolated(
             # `~/.config/pip/pip.conf`.
             pip_args.append("--isolated")
 
+        # Configure a keychain provider if so configured and the version of Pip supports the option.
+        # Warn the user if Pex cannot pass the `--keyring-provider` option and suggest a solution.
+        if package_index_configuration and package_index_configuration.keyring_provider:
+            if self.version.version >= PipVersion.v23_1.version:
+                pip_args.append("--keyring-provider")
+                pip_args.append(package_index_configuration.keyring_provider)
+            else:
+                warn_msg = textwrap.dedent(
+                    """
+                    The PIP_KEYRING_PROVIDER option is set, but Pip v{THIS_VERSION} does not support the `--keyring-provider` option
```

Comment:
@jsirois : Is there any additional context you would want in this warning message?

Reply:
Instead of `PIP_KEYRING_PROVIDER`, this warning should probably mention the Pex `--keyring-provider` option instead. This PR configures keyring provider based on that `--keyring-provider` option and not the environment variable.

Although, this begs the question, should Pex be looking in the environment for `PIP_KEYRING_PROVIDER` at all?

Reply:
It doesn't:
```
:; git grep PIP_KEYRING_PROVIDER
tests/integration/test_keyring_support.py:                PIP_KEYRING_PROVIDER="subprocess",
tests/integration/test_keyring_support.py:            env=make_env(PIP_KEYRING_PROVIDER="import", **devpi_clean_env),
```

And it shouldn't since `--use-pip-config` is a bit all bets off.

So I think you're talking to yourself here about the doc you added. I already agreed / pushed for `s/PEX_KEYRING_PROVIDER/--keyring-provider/` higher up in this conversation.

Reply:
Good point. Will update to `--keyring-provider`.

Reply:
Done.